### DISCUSSION TITLE
HiFi and HiC quality control before assembly

### DISCUSSION THREAD

**Discussion by YingChen94** [ERGA-consortium/pipelines#37] [General]
Hello,

Is it necessary to do QC on HiFi and HiC data before running them in HiFiasm? For example, shall I remove PCR duplicates in the HiC data? Do I need to remove contaminants before assembly or can I do it after assembly?

Any insights are appreciated! Thank you!
Ying

**Comment by diegomics** [comment]
Hi Ying,
It is always a good idea to check what the data looks like. Adapters are not always located in the same places in the HiFi reads, and in addition, the reads tend to be of lower quality, so it's better to remove the entire read instead. For the HiC reads, I advise doing a QC and trimming, too.

Please take a look at https://workflowhub.eu/collections/10

There, the two first workflows (WF0) are 'QC and trimming', next workflow (WF1) is for profiling, next is for Hifiasm assembly, etc.

You can take a look at the excellent tutorials from the VGP Galaxy team, too: https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html

Contamination screening is recommended after scaffolding. If you detected an important contamination problem in your reads during the QC and/or profiling steps, that's another story.

All the best ;)
Diego

**Reply by YingChen94** [reply]
Thank you for the suggestions Diego! They are very helpful!

**Comment by fka21** [comment]
I know this is a bit late, but I want to share something I ran into with QC before assembly. Filtering reads for contaminants with k-mer based approaches can introduce bias you don't expect.

I was using [Kraken2](https://github.com/DerrickWood/kraken2) and [KrakenTools](https://github.com/jenniferlu717/KrakenTools) to filter out bacterial and viral contaminants. The resulting assemblies were highly fragmented, even though every other QC metric said the input data was clean. After a lot of painful digging, I found the cause: reads were being flagged as bacterial when only a few k-mers matched bacteria and the rest were unclassified. Kraken2's fallback in that case is to call the whole read bacterial, even though the actual k-mer coverage is low.

So if your species of interest **isn't well represented in the k-mer database, I'd skip decontaminating the readset**.
